nflow-engine/src/main/java/io/nflow/engine/internal/executor/WorkflowDispatcher.java

```diff
@@ -80,6 +86,8 @@ public void run() {
           sleep(false);
         }
       }
+    } catch (IOException | ReflectiveOperationException e) {
```

Comment:
So my two commits just now added your better exception description, but otherwise the logic is the same.